src/main/java/digital/vix/vertxdemo/repository/InformationRepositoryImpl.java

```diff
@@ -0,0 +1,78 @@
+package digital.vix.vertxdemo.repository;
+
+import digital.vix.vertxdemo.models.Information;
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Single;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.sql.ResultSet;
+import io.vertx.reactivex.ext.sql.SQLClient;
+
+public class InformationRepositoryImpl implements InformationRepository {
+
+    private SQLClient sqlClient;
+
+    public InformationRepositoryImpl(SQLClient sqlClient) {
+        this.sqlClient = sqlClient;
+    }
+
+    @Override
+    public Flowable<JsonObject> all() {
+        return sqlClient.rxQuery("SELECT * FROM information")
+                .flatMapPublisher(resultSet -> Flowable.fromIterable(resultSet.getRows()));
+    }
+
+    @Override
+    public Single<ResultSet> findByEndPointId(long id) {
+        return sqlClient
+              .rxQueryWithParams("SELECT * FROM information where endpoint_id = ?", new JsonArray().add(id));
+    };
+
+    @Override
+    public Single<ResultSet> findById(long id) {
+        return sqlClient.rxQueryWithParams("SELECT * FROM information where id = ?", new JsonArray().add(id));
+    };
+
+
+    @Override
+    public Flowable<JsonObject> findByEndPointIds(long[] ids) {
+        JsonArray idsPara = new JsonArray();
+        String queryExtension ="";
+        for (long id:  ids){
+            idsPara.add(id);
+            queryExtension += "?, ";
+        }
+        queryExtension = (queryExtension.length()>0) ? queryExtension.substring(0,queryExtension.length()-2) : queryExtension;
+
+        return sqlClient
+                .rxQueryWithParams("SELECT * FROM information where endpoint_id IN (" + queryExtension+ ")= ?", idsPara)
```

Comment:
does it need the = ? at the end?
```suggestion
                .rxQueryWithParams("SELECT * FROM information where endpoint_id IN (" + queryExtension+ ")", idsPara)
```